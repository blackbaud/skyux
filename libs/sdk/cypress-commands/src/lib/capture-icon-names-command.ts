// eslint-disable-next-line @typescript-eslint/no-namespace
declare namespace Cypress {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  interface Chainable<Subject> {
    /**
     * Capture icon names that appear on the screen.
     */
    skyCaptureIconNames(): Chainable<void>;
  }
}

/**
 * Choose theme for visual regression testing.
 */
Cypress.Commands.add('skyCaptureIconNames', () => {
  const iconNames = Cypress.$('sky-icon-svg')
    .map((_idx, el) => el.getAttribute('data-sky-icon'))
    .get()
    .filter(Boolean) as string[];
  cy.task('skyCaptureIconNamesToFile', {
    test: Cypress.currentTest.title,
    file: Cypress.spec.name,
    iconNames,
  });
});
