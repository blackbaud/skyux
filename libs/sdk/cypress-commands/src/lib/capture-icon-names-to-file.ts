import fs from 'fs-extra';
import path from 'node:path';

export function captureIconNamesToFile(
  on: Cypress.PluginEvents,
  config: Cypress.PluginConfigOptions,
): void {
  on('task', {
    skyCaptureIconNamesToFile(data: {
      test: string;
      file: string;
      iconNames: string[];
    }) {
      if (data.iconNames.length > 0) {
        const outputPath = config.screenshotsFolder || config.downloadsFolder;
        const baseName =
          'sky-icon-names-' + path.basename(data.file).replaceAll('.', '-');
        const matchingFiles = fs
          .readdirSync(outputPath || '.')
          .filter((f) => f.startsWith(baseName) && f.endsWith('.json'));
        const suffix =
          matchingFiles.length > 0 ? `-${matchingFiles.length + 1}` : '';
        const outputFile = path.join(outputPath, `${baseName}${suffix}.json`);
        fs.writeJsonSync(outputFile, data, { spaces: 2 });
        console.log(`Captured icon names to ${outputFile}`);
      }
      return null;
    },
  });
}
