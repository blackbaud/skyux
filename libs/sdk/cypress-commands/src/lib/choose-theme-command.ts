// eslint-disable-next-line @typescript-eslint/no-namespace
declare namespace Cypress {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  interface Chainable<Subject> {
    /**
     * Choose theme for visual regression testing.
     */
    skyChooseTheme(theme: string): Chainable<void>;
  }
}

/**
 * Choose theme for visual regression testing.
 */
Cypress.Commands.add('skyChooseTheme', (theme: string) => {
  cy.get('select.sky-theme-selector')
    .first()
    .should('be.visible')
    .select(theme);
  const [themeName, themeMode] = theme.split('-');
  cy.get('body').should('have.class', `sky-theme-${themeName}`);
  cy.get('body').should('have.class', `sky-theme-mode-${themeMode || 'light'}`);
});
