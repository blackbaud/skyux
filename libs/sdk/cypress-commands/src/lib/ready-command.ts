// eslint-disable-next-line @typescript-eslint/no-namespace
declare namespace Cypress {
  interface Chainable<Subject> {
    skyReady(
      appName?: string,
      additionalExistingEls?: string[],
      additionalVisibleEls?: string[],
    ): Chainable<Subject>;
  }
}

Cypress.Commands.add(
  'skyReady',
  (
    appName?: string,
    additionalExistingEls?: string[],
    additionalVisibleEls?: string[],
  ) => {
    additionalExistingEls?.forEach((el) => {
      cy.get(el).should('exist');
    });

    additionalVisibleEls?.forEach((el) => {
      cy.get(el).should('exist').should('be.visible');
    });

    if (appName) {
      return cy
        .get('#assetsLoaded')
        .should('exist')
        .get(appName)
        .should('exist')
        .should('be.visible');
    } else {
      return cy.get('#assetsLoaded').should('exist');
    }
  },
);
