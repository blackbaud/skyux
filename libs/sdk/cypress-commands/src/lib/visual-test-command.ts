// eslint-disable-next-line @typescript-eslint/no-namespace
declare namespace Cypress {
  interface Chainable<Subject> {
    /**
     * Capture a screenshot of the current page for visual regression testing.
     */
    skyVisualTest(
      name: string,
      options?: Record<string, unknown>,
      selector?: string,
    ): Chainable<void>;
  }
}

/**
 * Capture a screenshot of the current page for visual regression testing. Include the URL in `blackout` because that is
 * the only arbitrary value field that becomes available in the after-screenshot hook, and it is required by Percy.
 */
Cypress.Commands.add(
  'skyVisualTest',
  {
    prevSubject: ['optional', 'element', 'window'],
  },
  (
    prevSubject: void | Window | Cypress.JQueryWithSelector<HTMLElement>,
    name: string,
    options?: Record<string, unknown>,
    selector?: string,
  ): void => {
    cy.url().then((url) => {
      const cyPrefix = selector ? cy.get(selector) : cy;

      if (
        prevSubject &&
        !Object.prototype.hasOwnProperty.call(prevSubject, 'window')
      ) {
        cyPrefix
          .wrap(prevSubject)
          .should('exist')
          .should('be.visible')
          .screenshot(name, {
            ...options,
            blackout: [`url:${url}`],
          });
      } else {
        cyPrefix.screenshot(name, {
          ...options,
          blackout: [`url:${url}`],
        });
      }
    });
  },
);
