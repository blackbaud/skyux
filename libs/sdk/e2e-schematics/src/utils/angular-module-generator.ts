import { Tree } from '@nx/devkit';
import { Schema as ComponentSchema } from '@schematics/angular/component/schema';
import { Schema as ModuleSchema } from '@schematics/angular/module/schema';

import { wrapAngularDevkitSchematic } from 'nx/src/adapter/ngcli-adapter';

export async function angularModuleGenerator(tree: Tree, schema: ModuleSchema) {
  return wrapAngularDevkitSchematic('@schematics/angular', 'module')(
    tree,
    schema
  );
}

export async function angularComponentGenerator(
  tree: Tree,
  schema: ComponentSchema
) {
  return wrapAngularDevkitSchematic('@schematics/angular', 'component')(
    tree,
    schema
  );
}
