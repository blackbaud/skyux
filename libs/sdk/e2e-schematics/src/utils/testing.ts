import {
  E2eTestRunner,
  UnitTestRunner,
  applicationGenerator,
  libraryGenerator,
} from '@nx/angular/generators';
import { Tree } from '@nx/devkit';
import { Linter } from '@nx/eslint';

export async function createTestApplication(
  tree: Tree,
  options: { name: string; e2eTestRunner?: boolean; unitTestRunner?: boolean },
): Promise<void> {
  await applicationGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `apps/${options.name}`,
    e2eTestRunner: options.e2eTestRunner
      ? E2eTestRunner.Cypress
      : E2eTestRunner.None,
    linter: Linter.None,
    unitTestRunner: options.unitTestRunner
      ? UnitTestRunner.Jest
      : UnitTestRunner.None,
  });
}

export async function createTestLibrary(
  tree: Tree,
  options: { name: string },
): Promise<void> {
  await libraryGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `libs/${options.name}`,
    linter: Linter.None,
    unitTestRunner: UnitTestRunner.None,
  });
}
