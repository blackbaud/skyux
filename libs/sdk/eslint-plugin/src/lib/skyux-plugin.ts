import all from './configs/all.json';
import skyNoId, { RULE_NAME as skyNoIdRuleName } from './rules/sky-no-id';

const skyuxPlugin = {
  meta: {
    name: '@skyux-sdk/eslint-plugin',
    version: '0.0.0-PLACEHOLDER',
  },
  configs: {
    all,
  },
  rules: {
    [skyNoIdRuleName]: skyNoId,
  },
  // processors: {},
};

export default skyuxPlugin;
