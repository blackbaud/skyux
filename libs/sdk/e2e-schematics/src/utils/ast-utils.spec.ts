import {
  applicationGenerator,
  componentGenerator,
} from '@nx/angular/generators';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import * as ts from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';

import { wrapAngularDevkitSchematic } from 'nx/src/adapter/ngcli-adapter';
import { createTree } from 'nx/src/generators/testing-utils/create-tree';

import { angularModuleGenerator } from './angular-module-generator';
import {
  applyTransformers,
  applyTransformersToPath,
  findComponentClass,
  findNgModuleClass,
  getInsertExportTransformer,
  getInsertIdentifierToArrayTransformer,
  getInsertImportTransformer,
  getInsertStringPropertyTransformer,
  getNamedImport,
  getRenameVariablesTransformer,
  getSourceAsString,
  getStringLiteral,
  getStringLiteralsSetterTransformer,
  getTransformerToAddExportToNgModule,
  readSourceFile,
  writeSourceFile,
} from './ast-utils';

describe('ast-utils', () => {
  it('should handle trying to read from an unknown file', () => {
    const tree = createTree();
    const returnedFile = readSourceFile(tree, 'fake/path');
    expect(returnedFile.fileName).toBe('fake/path');
    expect(returnedFile.getFullText()).toBe('');
  });

  it('should rename a variable', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(fileName, `const test: string = "value";\nconsole.log(test);\n`);
    const sourceFile = readSourceFile(tree, fileName);
    const transformer = getRenameVariablesTransformer({ test: 'update' });
    const [result] = applyTransformers([sourceFile], [transformer]);
    const resultCode = getSourceAsString(result);
    expect(resultCode).toEqual(
      `const update: string = "value";\nconsole.log(update);\n`
    );
    writeSourceFile(tree, fileName, result);
    expect(tree.read(fileName, 'utf-8')).toEqual(resultCode);
  });

  it('should getStringLiteral', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      `const test: string = "value";\nconst obj = { property: "prop" };\n`
    );
    const sourceFile = readSourceFile(tree, fileName);
    const value = getStringLiteral(sourceFile, 'test');
    expect(value).toEqual('value');
    const prop = getStringLiteral(sourceFile, 'property');
    expect(prop).toEqual('prop');
  });

  it('should getStringLiteral, throw error', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      `const test: string = "value";\nconst obj = { property: "prop" };\n`
    );
    const sourceFile = readSourceFile(tree, fileName);
    try {
      getStringLiteral(sourceFile, 'bogus');
      fail();
    } catch (e) {
      expect(e).toBeTruthy();
    }
  });

  it('should setStringLiterals', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      `const test: string = "value";\nconst obj = { property: "prop" };\n`
    );
    const transformer = getStringLiteralsSetterTransformer({
      test: 'new value',
      property: 'updated',
    });
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toEqual(
      `const test: string = "new value";\nconst obj = { property: "updated" };\n`
    );
  });

  it('should insertStringPropertyBefore', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(fileName, `const test = { test: "value", property: "prop" };\n`);
    const transformerTest = getInsertStringPropertyTransformer(
      'test',
      'id',
      'updated'
    );
    const transformerBogus = getInsertStringPropertyTransformer(
      'bogus',
      'added',
      'miss'
    );
    applyTransformersToPath(tree, fileName, [
      transformerTest,
      transformerBogus,
    ]);
    expect(tree.read(fileName, 'utf-8')).toEqual(
      `const test = { id: "updated", test: "value", property: "prop" };\n`
    );
  });

  it('should getInsertExportTransformer', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      [
        `export * from './lib/first.module'`,
        `export * from './lib/second.module'`,
        `export * from './lib/fourth.module'`,
      ].join('\n')
    );
    const transformer = getInsertExportTransformer(
      './lib/third.module',
      './lib/second.module'
    );
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();
  });

  it('should getInsertIdentifierToArrayTransformer', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      `
    export default {
      id: 'x-component',
      component: XComponent,
      decorators: [
        moduleMetadata({
          imports: [],
        }),
      ],
    }
    `
    );
    const transformer = getInsertIdentifierToArrayTransformer(
      'imports',
      'XModule'
    );
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();
  });

  it('should getNamedImport', () => {
    const importStatement = ts.createSourceFile(
      `script.ts`,
      `import { XComponent } from './lib/x-component';`,
      ts.ScriptTarget.Latest
    ).statements[0] as ts.ImportDeclaration;
    expect(getNamedImport(importStatement, 'XComponent')).toBeTruthy();
    const aliasImportStatement = ts.createSourceFile(
      `script.ts`,
      `import { XComponent as XCom } from './lib/x-component';`,
      ts.ScriptTarget.Latest
    ).statements[0] as ts.ImportDeclaration;
    expect(
      getNamedImport(aliasImportStatement, 'XComponent')?.name.text
    ).toEqual('XCom');
    const noNameImportStatement = ts.createSourceFile(
      `script.ts`,
      `import * from './lib/x-component';`,
      ts.ScriptTarget.Latest
    ).statements[0] as ts.ImportDeclaration;
    expect(() =>
      getNamedImport(noNameImportStatement, 'XComponent')
    ).toThrowError(
      `The import from ./lib/x-component does not have named imports.`
    );
    const nonImportStatement = ts.createSourceFile(
      `script.ts`,
      `var foo = 'bar';`,
      ts.ScriptTarget.Latest
    ).statements[0] as ts.ImportDeclaration;
    expect(() => getNamedImport(nonImportStatement, 'XComponent')).toThrowError(
      `Could not find an import.`
    );
  });

  it('should findNgModuleClass, no @angular/core', () => {
    const source = ts.createSourceFile(
      `script.ts`,
      `
    @NgModule({
      declarations: [XComponent],
    })
    export class XModule {}
    `,
      ts.ScriptTarget.Latest
    );
    expect(() => findNgModuleClass(source)).toThrowError(
      `Could not find @angular/core import.`
    );
  });

  it('should findNgModuleClass, no NgModule import', () => {
    const source = ts.createSourceFile(
      `script.ts`,
      `
    import { Component } from '@angular/core';
    @NgModule({
      declarations: [XComponent],
    })
    export class XModule {}
    `,
      ts.ScriptTarget.Latest
    );
    expect(() => findNgModuleClass(source)).toThrowError(
      `Could not find NgModule import.`
    );
  });

  it('should findNgModuleClass, no properties', () => {
    const source = ts.createSourceFile(
      `script.ts`,
      `
    import { NgModule } from '@angular/core';
    @NgModule()
    export class XModule {}
    `,
      ts.ScriptTarget.Latest
    );
    const ngModuleClass = findNgModuleClass(source);
    expect(ngModuleClass).toBeTruthy();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(Object.keys(ngModuleClass!.properties).length).toEqual(0);
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(ngModuleClass!.classDeclaration.name?.text).toEqual('XModule');
  });

  it('should findNgModuleClass', () => {
    const source = ts.createSourceFile(
      `script.ts`,
      `
    import { NgModule } from '@angular/core';
    @NgModule({
      declarations: [XComponent],
    })
    export class XModule {}
    `,
      ts.ScriptTarget.Latest
    );
    const ngModuleClass = findNgModuleClass(source);
    expect(ngModuleClass).toBeTruthy();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(ngModuleClass!.properties).toHaveProperty('declarations');
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(ngModuleClass!.classDeclaration.name?.text).toEqual('XModule');
  });

  it('should findComponentClass', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, { name: 'test' });
    await componentGenerator(tree, { name: 'test', project: 'test' });
    const componentClass = findComponentClass(
      readSourceFile(tree, 'apps/test/src/app/test/test.component.ts')
    );
    expect(componentClass).toBeTruthy();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(componentClass!.classDeclaration.name?.text).toEqual(
      'TestComponent'
    );
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(componentClass!.properties).toHaveProperty('templateUrl');
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(componentClass!.properties).toHaveProperty('styleUrls');
  });

  it('should findComponentClass, not component', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, { name: 'test' });
    await angularModuleGenerator(tree, { name: 'test', project: 'test' });
    await wrapAngularDevkitSchematic('@schematics/angular', 'pipe')(tree, {
      name: 'test/test',
      project: 'test',
      module: 'test',
    });
    const pipeClass = findComponentClass(
      readSourceFile(tree, 'apps/test/src/app/test/test.pipe.ts')
    );
    expect(pipeClass).toBeFalsy();
    const specClass = findComponentClass(
      readSourceFile(tree, 'apps/test/src/app/test/test.pipe.spec.ts')
    );
    expect(specClass).toBeFalsy();
  });

  it('should findComponentClass, component options not object', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, { name: 'test' });
    await componentGenerator(tree, { name: 'test', project: 'test' });
    tree.write(
      'apps/test/src/app/test/test.component.ts',
      `
      import { Component } from '@angular/core';
      @Component(false)
      export class TestComponent {}
    `
    );
    expect(() =>
      findComponentClass(
        readSourceFile(tree, 'apps/test/src/app/test/test.component.ts')
      )
    ).toThrowError(
      `The Component options for TestComponent are not an object literal`
    );
  });

  it('should findComponentClass, no decorated class', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, { name: 'test' });
    await componentGenerator(tree, { name: 'test', project: 'test' });
    tree.write(
      'apps/test/src/app/test/test.component.ts',
      `
      import { Component } from '@angular/core';
      export class TestComponent {}
    `
    );
    const componentClass = findComponentClass(
      readSourceFile(tree, 'apps/test/src/app/test/test.component.ts')
    );
    expect(componentClass).toBeFalsy();
  });

  it('should getInsertImportTransformer', () => {
    const tree = createTree();
    const fileName1 = 'script1.ts';
    const fileName2 = 'script2.ts';
    const fileName3 = 'script3.ts';
    tree.write(fileName1, `import { XModule } from './lib/x.module';`);
    tree.write(fileName2, `import { YModule } from './lib/y.module';`);
    tree.write(fileName3, `const variable = 'value';`);
    const transformer = getInsertImportTransformer('XModule', './lib/x.module');
    applyTransformersToPath(tree, fileName1, [transformer]);
    applyTransformersToPath(tree, fileName2, [transformer]);
    applyTransformersToPath(tree, fileName3, [transformer]);
    expect(tree.read(fileName1, 'utf-8')).toMatchSnapshot();
    expect(tree.read(fileName2, 'utf-8')).toMatchSnapshot();
    expect(tree.read(fileName3, 'utf-8')).toMatchSnapshot();
  });

  it('should getTransformerToAddExportToNgModule', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      `
      import { NgModule } from '@angular/core';
      @NgModule({})
      export class TestModule {}
    `
    );
    const transformer = getTransformerToAddExportToNgModule('TestComponent');
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();
  });

  it('should getTransformerToAddExportToNgModule, update', () => {
    const tree = createTree();
    const fileName = 'script.ts';
    tree.write(
      fileName,
      `
      import { NgModule } from '@angular/core';
      @NgModule({
        exports: [ExistingComponent],
      })
      export class TestModule {}
    `
    );
    const transformer = getTransformerToAddExportToNgModule('TestComponent');
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();
  });

  it('should getTransformerToAddExportToNgModule, no change', () => {
    const tree = createTree();
    const fileName = 'script.ts';

    // Already set.
    tree.write(
      fileName,
      `
      import { NgModule } from '@angular/core';
      @NgModule({
        exports: [TestComponent],
      })
      export class TestModule {}
    `
    );
    const transformer = getTransformerToAddExportToNgModule('TestComponent');
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();

    // Unexpected value.
    tree.write(
      fileName,
      `
      import { NgModule } from '@angular/core';
      @NgModule({
        exports: false,
      })
      export class TestModule {}
    `
    );
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();

    // Class not decorated.
    tree.write(
      fileName,
      `
      import { NgModule } from '@angular/core';
      export class TestModule {}
    `
    );
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();

    // No module import.
    tree.write(
      fileName,
      `
      export class TestModule {}
    `
    );
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();

    // No module decorator.
    tree.write(
      fileName,
      `
      import { Component } from '@angular/core';
      export class TestModule {}
    `
    );
    applyTransformersToPath(tree, fileName, [transformer]);
    expect(tree.read(fileName, 'utf-8')).toMatchSnapshot();
  });
});
