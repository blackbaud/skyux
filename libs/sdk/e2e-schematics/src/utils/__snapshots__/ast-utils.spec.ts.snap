// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ast-utils should getInsertExportTransformer 1`] = `
"export * from './lib/first.module';
export * from './lib/second.module';
export * from './lib/third.module';
export * from './lib/fourth.module';
"
`;

exports[`ast-utils should getInsertIdentifierToArrayTransformer 1`] = `
"export default {
    id: 'x-component',
    component: XComponent,
    decorators: [
        moduleMetadata({
            imports: [XModule],
        }),
    ],
};
"
`;

exports[`ast-utils should getInsertImportTransformer 1`] = `
"import { XModule } from './lib/x.module';
"
`;

exports[`ast-utils should getInsertImportTransformer 2`] = `
"import { XModule } from './lib/x.module';
import { YModule } from './lib/y.module';
"
`;

exports[`ast-utils should getInsertImportTransformer 3`] = `
"import { XModule } from './lib/x.module';
const variable = 'value';
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule 1`] = `
"import { NgModule } from '@angular/core';
@NgModule({ exports: [TestComponent] })
export class TestModule {
}
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule, no change 1`] = `
"import { NgModule } from '@angular/core';
@NgModule({
    exports: [TestComponent],
})
export class TestModule {
}
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule, no change 2`] = `
"import { NgModule } from '@angular/core';
@NgModule({
    exports: false,
})
export class TestModule {
}
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule, no change 3`] = `
"import { NgModule } from '@angular/core';
export class TestModule {
}
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule, no change 4`] = `
"export class TestModule {
}
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule, no change 5`] = `
"import { Component } from '@angular/core';
export class TestModule {
}
"
`;

exports[`ast-utils should getTransformerToAddExportToNgModule, update 1`] = `
"import { NgModule } from '@angular/core';
@NgModule({
    exports: [ExistingComponent, TestComponent],
})
export class TestModule {
}
"
`;
